tests/containers/slem_rancher.pm

```diff
@@ -0,0 +1,46 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Test SLE Micro Rancher image
+#   This image is used as a base to build a Rancher Harverster container image.
+#   Then, that image will be used to build a Host OS on top, so
+#   it includes the kernel, firmware, bootloader, etc.
+#
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use Mojo::Base qw(consoletest);
+use testapi;
+use utils;
+
+sub run {
+    my ($self) = @_;
+    $self->select_serial_terminal;
+
+    my $image = get_var('CONTAINER_IMAGE_TO_TEST', 'registry.suse.de/suse/sle-15-sp3/update/products/microos52/update/cr/totest/images/suse/sle-micro-rancher/5.2:latest');
+
+    assert_script_run("podman pull $image");
+    assert_script_run("podman run --name slem_image -dt $image");
+
+    record_info('Kernel', 'Test that kernel files are present');
+    validate_script_output("podman exec slem_image /bin/sh -c 'ls /boot'", sub { /initrd/ });
+    validate_script_output("podman exec slem_image /bin/sh -c 'ls /boot'", sub { /vmlinuz/ });
+
+    record_info('Firmware', 'Test that /lib/firmware directory is not empty');
+    assert_script_run("podman exec slem_image /bin/sh -c 'test -d /lib/firmware'");
+    validate_script_output("podman exec slem_image /bin/sh -c 'rmdir /lib/firmware/' 2>&1", sub { /Directory not empty/ }, proceed_on_failure => 1);
+
+    record_info('grub', 'Test that /etc/default/grub exists');
+    assert_script_run("podman exec slem_image /bin/sh -c 'test -f /etc/default/grub'");
+
+    record_info('repos', 'Image should come with empty repos');
+    validate_script_output("podman exec slem_image /bin/sh -c 'zypper lr' 2>&1", sub { /No repositories defined/ }, proceed_on_failure => 1);
+    assert_script_run("podman exec slem_image /bin/sh -c 'rmdir /etc/zypp/repos.d'");
```

Comment:
`ls -A` would return nothing, if the directory is empty. Maybe that might be more helpful when some repo would leak there accidentally.

Reply:
yes, but then I'd need to use a comparison that `ls -A` doesn't return anything. rmdir will immediatelly fail if there is something in this dir.

Reply:
yup, that would require another `validate_script_output`

Reply:
ah my previous comment was here commented

Reply:
Please check now, following Ivan's suggestion.

Reply:
Me gusta! Buenisimo :)